cinema/views.py

```diff
@@ -21,19 +23,29 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(mixins.ListModelMixin,
+                   mixins.CreateModelMixin,
+                   viewsets.GenericViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
The `GenreViewSet` correctly uses the `IsAdminOrIfAuthenticatedReadOnly` permission class, which aligns with the task requirement to limit actions based on user roles.